test/utils.ts

```diff
@@ -66,3 +69,106 @@ export async function ensureEditorServicesIsConnected(): Promise<IPowerShellExte
     extension.unregisterExternalExtension(sessionId);
     return extension;
 }
+
+/**
+ * This is a workaround for sinon not being able to stub interfaces. Interfaces are a TypeScript-only concept so this effectively allows us to stub interfaces by not providing the entire implementation but only what matters for the test. "What matters" is not type checked so you must be careful to stub everything you need, otherwise you should provide a default implementation instead if you do not know.
+*/
+export function stubInterface<T>(object?: Partial<T>): T {
+    return object ? object as T : {} as T;
+}
+
+/** Installs the C# extension in "this" running vscode version for binary module debug testing */
+export async function InstallCSharpExtension(): Promise<void> {
+    // Install the csharp extension which is required for the dotnet debugger testing
+    if (!vscode.extensions.getExtension("ms-dotnettools.csharp")) {
+        //HACK: There is no way to set the initial cwd using RunTests, and resolveCliArgsFromVSCodeExecutablePath requires the cwd to be the extension root, so we set it here temporarily: https://github.com/microsoft/vscode-test/blob/addc23e100b744de598220adbbf0761da870eda9/lib/download.ts#L31
+        const extensionRootPath = path.resolve(`${__dirname}/../..`);
+        if (!existsSync(path.join(extensionRootPath, "package.json"))) {
+            throw new Error(`Expected to find package.json at ${extensionRootPath}. Did you move utils.ts out of the test folder?`);
+        }
+        const cwd = process.cwd();
+        process.chdir(extensionRootPath);
+
+        //HACK: On Mac, resolveCliArgs expects the electron path but we are currently running in a special helper child process, so we need to find electron: https://github.com/microsoft/vscode-test/blob/addc23e100b744de598220adbbf0761da870eda9/lib/util.ts#L158
+
+        console.log("==INSTALLPREREQUISITE: C# Extension==");
+        const codeExePath = process.platform === "darwin" && process.execPath.endsWith("Helper (Plugin)")
+            ? path.join(path.dirname(process.execPath), "../../../../MacOS", "Electron")
+            : process.execPath;
```

Comment:
Gosh that looks annoying to have figured out.

Reply:
Figuring it out wasn't that difficult because i was already looking at execPath, but I really don't like this flaky workaround. I have some ideas there for later but this is good enough for now.